spectrocloud/resource_cluster_aws.go

```diff
@@ -469,90 +480,93 @@ func resourceClusterAwsUpdate(ctx context.Context, d *schema.ResourceData, m int
-						return diag.FromErr(err)
-					}
+			nsMap := make(map[string]interface{})
+
+			for _, mp := range ns.List() {
+				machinePoolResource := mp.(map[string]interface{})
+				nsMap[machinePoolResource["name"].(string)] = machinePoolResource
+				// since known issue in TF SDK: https://github.com/hashicorp/terraform-plugin-sdk/issues/588
+				if machinePoolResource["name"].(string) != "" {
+					name := machinePoolResource["name"].(string)
+					if name != "" {
+						hash := resourceMachinePoolAwsHash(machinePoolResource)
+						vpcId := d.Get("cloud_config").([]interface{})[0].(map[string]interface{})["vpc_id"]
 
-					if oldMachinePool, ok := osMap[name]; !ok {
-						log.Printf("Create machine pool %s", name)
-						err = c.CreateMachinePoolAws(cloudConfigId, machinePool, ClusterContext)
-					} else if hash != resourceMachinePoolAwsHash(oldMachinePool) {
-						log.Printf("Change in machine pool %s", name)
-						err = c.UpdateMachinePoolAws(cloudConfigId, machinePool, ClusterContext)
-						// Node Maintenance Actions
-						err := resourceNodeAction(c, ctx, nsMap[name], c.GetNodeMaintenanceStatusAws, CloudConfig.Kind, ClusterContext, cloudConfigId, name)
+						var err error
+						machinePool, err := toMachinePoolAws(machinePoolResource, vpcId.(string))
 						if err != nil {
 							return diag.FromErr(err)
 						}
-					}
 
-					if err != nil {
-						return diag.FromErr(err)
-					}
+						if oldMachinePool, ok := osMap[name]; !ok {
+							log.Printf("Create machine pool %s", name)
+							err = c.CreateMachinePoolAws(cloudConfigId, machinePool, ClusterContext)
+						} else if hash != resourceMachinePoolAwsHash(oldMachinePool) {
+							log.Printf("Change in machine pool %s", name)
+							err = c.UpdateMachinePoolAws(cloudConfigId, machinePool, ClusterContext)
+							// Node Maintenance Actions
+							err := resourceNodeAction(c, ctx, nsMap[name], c.GetNodeMaintenanceStatusAws, CloudConfig.Kind, ClusterContext, cloudConfigId, name)
+							if err != nil {
+								return diag.FromErr(err)
+							}
+						}
 
-					// Processed (if exists)
-					delete(osMap, name)
+						if err != nil {
+							return diag.FromErr(err)
+						}
+
+						// Processed (if exists)
+						delete(osMap, name)
+					}
 				}
 			}
-		}
 
-		// Deleted old machine pools
-		for _, mp := range osMap {
-			machinePool := mp.(map[string]interface{})
-			name := machinePool["name"].(string)
-			log.Printf("Deleted machine pool %s", name)
-			if err := c.DeleteMachinePoolAws(cloudConfigId, name, ClusterContext); err != nil {
-				return diag.FromErr(err)
+			// Deleted old machine pools
+			for _, mp := range osMap {
+				machinePool := mp.(map[string]interface{})
+				name := machinePool["name"].(string)
+				log.Printf("Deleted machine pool %s", name)
+				if err := c.DeleteMachinePoolAws(cloudConfigId, name, ClusterContext); err != nil {
+					return diag.FromErr(err)
+				}
 			}
 		}
 
+		diagnostics, done := updateCommonFields(d, c)
```

Comment:
One more thing here - I guess it's a long pending change, let's refactor at some point to return error instead of bool @SivaanandM

Reply:
sure will discuss and update it